### PR TITLE
Notes for added/removed labels to/from an issue.

### PR DESCRIPTION
**Pull request by PaulaCara**
The idea of this feature is to create a note each time a label is added/removed to/from an issue.

Implements feature request:  http://feedback.gitlab.com/forums/176466-general/suggestions/6623258-create-a-note-when-issue-labels-change

Screenshots:
screenshot from 2014-11-04 11 00 48
screenshot from 2014-11-04 11 01 12

### CODE REVIEW

**Comment by PaulaCara** [comment]
@dblessing I will add the same functionality to merge requests as soon as I can.

File: app/services/merge_requests/update_service.rb [hunk 1]
```diff
@@ -23,11 +23,15 @@ def execute(merge_request)
         merge_request.update_nth_task(params[:task_num].to_i, false)
       end
 
+      old_labels = merge_request.labels.map{ |label| label }
```
Comment (dblessing) [inline app/services/merge_requests/update_service.rb]: This should also just be `merge_request.labels`. No need for map, it's just creating an array again.

**Comment by dblessing** [comment]
@PaulaCara The merge request stuff is looking good. Please fix the hound warnings and the few comments I made. Also, please add merge request tests.

File: app/services/merge_requests/update_service.rb (hunk 1 above)
Comment (PaulaCara) [inline app/services/merge_requests/update_service.rb]: If I don't create an array again, `old_labels` will hold a reference to `merge_request.labels` and will be the same as the new labels after the attributes update.

Comment (dblessing) [inline app/services/merge_requests/update_service.rb]: What about `merge_request.labels.to_a`? That will remove the reference to the actual collection and it won't be changed.

**Comment by nikashitsa** [comment]
:+1: 

@PaulaCara nice feature. Are you going to finish it?

**Comment by PaulaCara** [comment]
@nikashitsa I will finish the feature as soon as I have time.

**Comment by nikashitsa** [comment]
@PaulaCara can I help you?  :smirk:

**Comment by nikashitsa** [comment]
Can't wait until this will be done. @dblessing please have a look at my [improved version] (https://gitlab.com/gitlab-org/gitlab-ce/merge_requests/331).

**Comment by Razer6** [comment]
@nikashitsa Thank you and @PaulaCara for the initial PR

**Pull request closed without merging**